File: Frontend/models/getter/group/GroupViewModel.ts
import GroupUserViewModel from './GroupUserViewModel';
import GroupItemViewModel from './GroupItemViewModel';

export default class GroupViewModel {
  public Id: number;
  public Name: string;
  public Users: Array<GroupUserViewModel> = [];
  public Items: Array<GroupItemViewModel> = [];

  constructor(data?) {
    if (!data) return;
    this.Id = data.id;
    this.Name = data.name;
    this.Users = data.users.map(u => new GroupUserViewModel(u));
    this.Items = data.items.map(i => new GroupItemViewModel(i));
  }
}

File: Backend/Application/Groups/Queries/GroupDetail/GroupViewModel.cs
using System.Collections.Generic;
using System.Linq;
using Domain.Models;

namespace Application.Groups.Queries.GroupDetail
{
    public class GroupViewModel
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public IEnumerable<GroupUserViewModel> Users { get; set; }
        public IEnumerable<GroupItemViewModel> Items { get; set; }

        public GroupViewModel(IReadOnlyCollection<UserGroup> userGroups)
        {
            var group = userGroups.First().Group;
            Id = group.Id;
            Name = group.Name;
            Users = userGroups
                .Select(ug => new GroupUserViewModel(ug))
                .OrderByDescending(u => u.IsOwner)
                .ThenByDescending(u => u.IsManager)
                .ThenByDescending(u => u.IsAcceptedByManager)
                .ThenByDescending(u => u.IsAcceptedByUser);
            Items = group.GroupItems.Select(gi => new GroupItemViewModel(gi))
                .OrderBy(gi => gi.Name);
        }
    }
}